### DISCUSSION TITLE
Is it possible to extend a DatasourceWriter or manually select one?

### DISCUSSION THREAD

**Discussion by rjborba** [benyap/firestore-backfire#410] [Q&A] [answered]
Hey! First of all thanks for this lib. 

I'm trying to upload my backup to DigitalOcean Spaces. It claims to be compatible with AWS S3. I was wondering if s3 writer could be able to do the operations with DigitalOcean address. But seems like the matcher defaults to local. If not possible to just reuse s3 writer, is there a way to extend it?

**Comment by benyap** [comment, marked as answer]
Hey @rjborba! Yes, it is possible to use your own implementations of a data writer and reader. Please take a look at the documentation for it here: https://github.com/benyap/firestore-backfire#custom-data-sources

The limitation here is that you'll need to run the program using Node. You won't be able to use custom data sources directly on the CLI.

Feel free to add to this discussion or create a new topic if you have more questions about how to do this 😊
